### PR TITLE
 feat: add course team instructor management filters, views and backends

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds the necessary implementations for managing course team instructors, which are:
- A filter that adds a new instructor tab to the instructor dashboard for: adding, viewing and removing CourseTeamsInsturctor records. This filter adds the tab for users with instructor roles (course admin in the team membership tab)
- A filter that modifies which instructor tabs are shown to users with limited staff-level access.
-  An authentication backend that manages the user access at a permission level. 
 
So, a course team instructor is a user with limited staff access level associated with a team through CourseTeamsInstrucctor by an instructor.

### How to test
1. Install this plugin using this branch.
2. To add the filters, please add the following config to your LMS env settings:
```
    OPEN_EDX_FILTERS_CONFIG = {
        "org.openedx.learning.instructor.dashboard.render.started.v1": {
            "fail_silently": False,
            "pipeline": [
                "openedx_unidigital.filters.TeamLimitedStaffDashboard",
                "openedx_unidigital.filters.TeamAssignmentDashboard",
            ]
        },
    }
```
3. Since we're working with teams, I suggest you create a team-set with two teams.
4. Give course admin access to a user, and limited staff access to a different one. We'll use this as our tests subjects.
5. As the course admin, access the instructor dashboard. You should be able to see the Courses' Team Instructor management tab. 
6. Now, as a student, join a team.
7. As the course admin, try the Student Admin tab. You should be able to do all operations.
8. As the course limited staff, try the Student Admin tab. You should get 200 for all operations.
9. In the instructor dashboard, as the course admin, associate the limited staff user to a different team the users belong to. Then, try again performing the operations from the Student Admin as the limited staff user. You should get 403 for all operations. 
19. Now, as the course admin, remove the limited staff from that team in the new instructor dashboard management tab and add them to the same team as the users. You should be able to perform all operations.

For more info on: 
- When the auth backend allows access
- When the new tab for course teams instructors management appears
- When the tabs are filtered to show only Course Info and Student admin

Refer to the documentation: docs/2-assign-instructors-to-course-teams.rst

### CODE REVIEW

**Review by BryanttV — COMMENTED** [review]
Hi @mariajgrimaldi, I was doing some tests locally and found the following:
1. In the drop-down list all the teams in the course are displayed, but in case there are two teams with the same name there is no way to differentiate them. Would it be useful to include the topic/group to which each team belongs?
image
2. Should we validate that the user exists?
image

File: openedx_unidigital/static/js/course_team_management.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    .then(function (data) {
+      addInstructorToTable(data);
+      $("#add-instructor-input").val("");
+    });
+});
+
+$("#course-teamset").on("change", function (select) {
+  const teamId = $("#course-teamset").val();
+  var instructorsTable = $("#instructors-table");
+  var instructorsTableBody = $("#instructors-table-body");
```
Comment (BryanttV) [inline openedx_unidigital/static/js/course_team_management.js]: Unused variable

File: openedx_unidigital/static/js/course_team_management.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      if (cookie.substring(0, name.length + 1) === name + "=") {
+        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+        break;
+      }
+    }
+  }
+  return cookieValue;
+}
+function addInstructorToTable(data) {
+  var instructorsTableBody = $("#instructors-table-body");
```
Comment (BryanttV) [inline openedx_unidigital/static/js/course_team_management.js]: We should use `let` instead of `var`

Comment (mariajgrimaldi) [reply openedx_unidigital/static/js/course_team_management.js]: Done! Thank you

File: openedx_unidigital/static/js/course_team_management.js (hunk 1 above)
Comment (mariajgrimaldi) [reply openedx_unidigital/static/js/course_team_management.js]: Done! Thanks

**Comment by mariajgrimaldi** [comment]
@BryanttV I implemented both of your suggestions. Could you check again? Thanks!

**Review by BryanttV — APPROVED** [review]
Thank you for addressing my comments! LGTM! :rocket:

**Pull request merged**